### PR TITLE
Search tests

### PR DESCRIPTION
**Pull request by Vladyslav1992**
Please, approve PR with Search tests

### CODE REVIEW

File: SimpleAutomationCommon/Pages/BasePage.cs [hunk 1]
```diff
@@ -21,26 +23,17 @@ public class BasePage<TOwner> : Page<TOwner>
         [FindByXPath("//img[@alt = 'SimplCommerce']")]
         public Label<TOwner> Logo { get; set; }
 
-        [FindByCss("input.form-control")]
-        public Label<TOwner> Search { get; set; }
+        [FindByCss("div.js-show-modal-search")]
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/BasePage.cs]: div[class*='search']

File: SimpleAutomationCommon/Pages/BasePage.cs [hunk 2]
```diff
@@ -21,26 +23,17 @@ public class BasePage<TOwner> : Page<TOwner>
...
+        public Clickable<TOwner> OpenSearchButton { get; set; }
 
-        [FindByCss("select.form-control")]
-        public Label<TOwner> CategorySelector { get; set; }
+        [FindByCss("button.btn-hide-modal-search")]
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/BasePage.cs]: button[class*='search']

File: SimpleAutomationCommon/Pages/BasePage.cs [hunk 3]
```diff
@@ -72,10 +65,25 @@ public class BasePage<TOwner> : Page<TOwner>
         [FindByXPath("//p[text()='Powered by ']")]
         public Label<TOwner> PoweredBy { get; set; }
 
+        [FindByCss("small.badge-results")]
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/BasePage.cs]: findByClassname('badge-results')

File: SimpleAutomationCommon/Pages/BasePage.cs [hunk 4]
```diff
@@ -72,10 +65,25 @@ public class BasePage<TOwner> : Page<TOwner>
...
         public string GetUser()
         {
             var userName = FullName.Get();
             return userName.Remove("Hello ", "!");
         }
+
+        public string GetResult()
+        {
+            var result = Results.Get().SplitIntoWords().ToList();
+            return result.Last();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/BasePage.cs]: transfer this to expression body   =>Results.Get().SplitIntoWords().Last();

File: SimpleAutomationCommon/Pages/BasePage.cs [hunk 5]
```diff
@@ -72,10 +65,25 @@ public class BasePage<TOwner> : Page<TOwner>
...
+        }
+
+        public int GetCountOfResults()
+        {
+            var result = Results.Get().SplitIntoWords().ToList();
+            return result.First().ToInt();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/BasePage.cs]: the same here =>

File: SimpleAutomationCommon/Pages/HomePage.cs [hunk 6]
```diff
@@ -0,0 +1,27 @@
+﻿using Atata;
+
+namespace SimpleAutomationCommon.Pages
+{
+    using _ = HomePage;
+
+    public class HomePage : BasePage<_>
+    {
+        [FindByXPath("//ul[@class='main-menu']//a[@href='/woman']")]
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/HomePage.cs]: can simplify to  - //a[@href='/woman'] ?

File: SimpleAutomationCommon/Pages/HomePage.cs [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        public Link<WomanPage, _> Woman { get; set; }
+
+        [FindByXPath("//ul[@class='main-menu']//a[@href='/man']")]
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/HomePage.cs]: same

File: SimpleAutomationCommon/Pages/HomePage.cs [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        public Link<ManPage, _> Man { get; set; }
+
+        [FindByXPath("//a[@href='/t-shirt']")]
+        public Link<TshirtPage, _> Tshirt { get; set; }
+
+        [FindByXPath("//a[@href='/suit']")]
+        public Link<SuitPage, _> Suit { get; set; }
+
+        [FindByXPath("//ul[@class='main-menu']//a[@href='/shoes']")]
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/HomePage.cs]: same

File: SimpleAutomationCommon/Pages/HomePage.cs [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        public Link<ShoesPage, _> Shoes { get; set; }
+
+        [FindByXPath("//ul[@class='main-menu']//a[@href='/watches']")]
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/HomePage.cs]: same

File: SimpleAutomationCommon/Pages/SearchPage.cs [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+        {
+            public H5<_> ProductName { get; set; }
+        }
+
+        [FindByCss("h2")]
+        private H2<_> SearchTitle { get; set; }
+
+        public string GetSearchKeywordFromTitle()
+        {
+            var keyword = SearchTitle.Get().SplitIntoWords().ToList().Last();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/SearchPage.cs]: transfer to expression body type =>

File: SimpleAutomationCommon/Pages/SearchPage.cs [hunk 11]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Linq;
+using Atata;
+
+namespace SimpleAutomationCommon.Pages
+{
+    using _ = SearchPage;
+
+    [Url("search")]
+    public class SearchPage : BasePage<_>
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/SearchPage.cs]: reorder methods and properties in this class - sort it

File: SimpleAutomationCommon/Pages/SearchPage.cs [hunk 12]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Linq;
+using Atata;
+
+namespace SimpleAutomationCommon.Pages
+{
+    using _ = SearchPage;
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/SearchPage.cs]: move using to one place

File: SimpleAutomationCommon/Pages/SuitPage.cs [hunk 13]
```diff
@@ -0,0 +1,18 @@
+﻿using Atata;
+
+namespace SimpleAutomationCommon.Pages
+{
+    using _= SuitPage;
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/SuitPage.cs]: same

File: SimpleAutomationCommon/Pages/TshirtPage.cs [hunk 14]
```diff
@@ -0,0 +1,18 @@
+﻿using Atata;
+
+namespace SimpleAutomationCommon.Pages
+{
+    using _ = TshirtPage;
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/TshirtPage.cs]: same

File: SimpleAutomationCommon/Pages/SuitPage.cs [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+
+    [Url("suit")]
+    public class SuitPage : BasePage<_>
+    {
+        [FindByCss("h2")]
+        private H2<_> SuitTitle { get; set; }
+
+        public string GetTitle()
+        {
+            return SuitTitle.Get();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/SuitPage.cs]: to =>

File: SimpleAutomationCommon/Pages/TshirtPage.cs [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+
+    [Url("t-shirt")]
+    public class TshirtPage : BasePage<_>
+    {
+        [FindByCss("h2")]
+        private H2<_> TshirtTitle { get; set; }
+
+        public string GetTitle()
+        {
+            return TshirtTitle.Get();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationCommon/Pages/TshirtPage.cs]: to =>

File: SimpleAutomationTests/Tests/SearchTest.cs [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+namespace SimpleAutomationTests.Tests
+{
+    public class SearchTest : BaseTest
+    {
+        [Test]
+        public void WemanClothesSearch()
+        {
+            var womanPage = Go.To<HomePage>().Woman.ClickAndGo();
+
+            var title = womanPage.GetTitle();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationTests/Tests/SearchTest.cs]: to =>

File: SimpleAutomationTests/Tests/SearchTest.cs [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+            title.Should().Be("Woman");
+
+            var result = womanPage.GetResult();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationTests/Tests/SearchTest.cs]: same

File: SimpleAutomationTests/Tests/SearchTest.cs [hunk 19]
```diff
@@ -0,0 +1,163 @@
...
+            result.Should().Be("results");
+        }
+
+        [Test]
+        public void MenClothesSearch()
+        {
+            var manPage = Go.To<HomePage>().Man.ClickAndGo();
+
+            var title = manPage.GetTitle();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationTests/Tests/SearchTest.cs]: same

File: SimpleAutomationTests/Tests/SearchTest.cs [hunk 20]
```diff
@@ -0,0 +1,163 @@
...
+            title.Should().Be("Man");
+
+            var result = manPage.GetResult();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationTests/Tests/SearchTest.cs]: same

File: SimpleAutomationTests/Tests/SearchTest.cs [hunk 21]
```diff
@@ -0,0 +1,163 @@
...
+            result.Should().Be("results");
+        }
+
+        [Test]
+        public void ShoesSearch()
+        {
+            var shoesPage = Go.To<HomePage>().Shoes.ClickAndGo();
+
+            var title = shoesPage.GetTitle();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationTests/Tests/SearchTest.cs]: same

File: SimpleAutomationTests/Tests/SearchTest.cs [hunk 22]
```diff
@@ -0,0 +1,163 @@
...
+            result.Should().Be("results");
+        }
+
+        [Test]
+        public void WatchesSearch()
+        {
+            var watchesPage = Go.To<HomePage>().Whatches.ClickAndGo();
+
+            var title = watchesPage.GetTitle();
+            title.Should().Be("Watches");
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationTests/Tests/SearchTest.cs]: same

File: SimpleAutomationTests/Tests/SearchTest.cs [hunk 23]
```diff
@@ -0,0 +1,163 @@
...
+
+            var result = watchesPage.GetResult();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationTests/Tests/SearchTest.cs]: same

File: SimpleAutomationTests/Tests/SearchTest.cs [hunk 24]
```diff
@@ -0,0 +1,163 @@
...
+
+        [Test]
+        public void MenTshirtsSearch()
+        {
+            var manSubMenu = Go.To<HomePage>().Man.Hover();
+            manSubMenu.Tshirt.IsVisible.Should.BeTrue();
+
+            var tshirtPage = manSubMenu.Tshirt.ClickAndGo();
+
+            var title = tshirtPage.GetTitle();
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationTests/Tests/SearchTest.cs]: omg! same!

File: SimpleAutomationTests/appsettings.json [hunk 25]
```diff
@@ -1,7 +1,7 @@
 {
-  "App": {
-    "homeUrl": "http://super.mega.portal:5000/"
-  },
+	"App": {
```
Comment (ViacheslavMelnichenko) [inline SimpleAutomationTests/appsettings.json]: why? cancel this

File: SimpleAutomationCommon/Pages/HomePage.cs (hunk 6 above)
Comment (Vladyslav1992) [reply SimpleAutomationCommon/Pages/HomePage.cs]: No, there are several elements with the same locator on the page.

File: SimpleAutomationCommon/Pages/HomePage.cs (hunk 7 above)
Comment (Vladyslav1992) [reply SimpleAutomationCommon/Pages/HomePage.cs]: No, there are several elements with the same locator on the page.

File: SimpleAutomationCommon/Pages/HomePage.cs (hunk 8 above)
Comment (Vladyslav1992) [reply SimpleAutomationCommon/Pages/HomePage.cs]: No, there are several elements with the same locator on the page.

File: SimpleAutomationCommon/Pages/HomePage.cs (hunk 9 above)
Comment (Vladyslav1992) [reply SimpleAutomationCommon/Pages/HomePage.cs]: No, there are several elements with the same locator on the page.

File: SimpleAutomationTests/appsettings.json (hunk 25 above)
Comment (Vladyslav1992) [reply SimpleAutomationTests/appsettings.json]: Didn't get what is wrong(

**Pull request merged**